modules/e2ee/OlmAdapter.js

```diff
@@ -470,6 +572,199 @@ export class OlmAdapter extends Listenable {
+
+                if (keySent) {
+                    return;
+                }
+
+                const ack = {
+                    [JITSI_MEET_MUC_TYPE]: OLM_MESSAGE_TYPE,
+                    olm: {
+                        type: OLM_MESSAGE_TYPES.SAS_KEY,
+                        data: {
+                            key: pubKey,
+                            uuid
+                        }
+                    }
+                };
+
+                this._sendMessage(ack, pId);
+
+                olmData.sasVerification.keySent = true;
+            } else {
+                logger.debug(`Received sas key message from ${pId} but we have no session for them!`);
+
+                this._sendError(participant, 'No session found while processing sas-key');
+            }
+            break;
+        }
+        case OLM_MESSAGE_TYPES.SAS_MAC: {
+            if (olmData.session) {
+                const { keys, mac, uuid } = msg.data;
+
+                if (!mac || !keys) {
+                    logger.warn('Invalid SAS MAC message');
+
+                    return;
+                }
+
+                const { sas, ed25519 } = olmData.sasVerification;
+
+                // Verify the received MACs.
+                const baseInfo = `${OLM_KEY_VERIFICATION_MAC_INFO}${pId}${this.myId}${uuid}`;
+                const keysMac = sas.calculate_mac(
+                    Object.keys(mac).sort().join(','), // eslint-disable-line newline-per-chained-call
+                    baseInfo + OLM_KEY_VERIFICATION_MAC_KEY_IDS
+                );
+
+                if (keysMac !== keys) {
+                    logger.error('SAS verification error: keys MAC mismatch');
+                    this.eventEmitter.emit(
+                        OlmAdapterEvents.PARTICIPANT_VERIFICATION_COMPLETED,
+                        pId,
+                        false,
+                        'keys MAC mismatch');
+
+                    return;
+                }
+
+                for (const [ keyInfo, computedMac ] of Object.entries(mac)) {
+                    const ourComputedMac = sas.calculate_mac(
+                        ed25519,
+                        baseInfo + keyInfo
+                    );
+
+                    if (computedMac !== ourComputedMac) {
+                        logger.error('SAS verification error: MAC mismatch');
+                        this.eventEmitter.emit(
+                            OlmAdapterEvents.PARTICIPANT_VERIFICATION_COMPLETED,
+                            pId,
+                            false,
+                            'MAC mismatch');
+
+                        return;
+                    }
+                }
+
+                logger.info(`SAS MAC verified for participant ${pId}`);
+                this.eventEmitter.emit(OlmAdapterEvents.PARTICIPANT_VERIFICATION_COMPLETED, pId, true);
+            } else {
+                logger.debug(`Received sas mac message from ${pId} but we have no session for them!`);
```

Comment:
Reverse the check, so the majority of the function has one level on indentation less